### PR TITLE
fix: Use model revision in results folder

### PR DESCRIPTION
**Pull request by isaac-chung**
Reason for dataset addition: Addresses #618 

## Changes
New results structure would incorporate model revisions like so:
```
results
|-- model name
  |-- model revision
    |-- datasetA.json
```

If the model revision cannot be found in the model (e.g. not loaded from huggingface), then it needs to be supplied to the `run` method or via the command line arg `--model_revision`.

## Test
Added a test run result with:
```
mteb -m intfloat/multilingual-e5-base -t BlurbsClusteringS2S.v2
```
where the latest revision of the [model](https://huggingface.co/intfloat/multilingual-e5-base/tree/main) is `d13f1b27baf31030b7fd040960d60d909913633f`.

### CODE REVIEW

**Review by KennethEnevoldsen — APPROVED** [review]
We might want to integrate it with #807 as well. I believe we can now leave get_model_meta() to extract metadata for an arbitrary sentence trf model.

File: mteb/cmd.py [hunk 1]
```diff
@@ -118,6 +118,14 @@ def main():
         help="Display the available tasks",
     )
 
+    ## model revision
+    parser.add_argument(
```
[the remark this answers is not in the document]
Comment (isaac-chung) [reply mteb/cmd.py:122]: Now loading specified revision

File: mteb/cmd.py
[the remark this answers is not in the document]
Comment (isaac-chung) [reply mteb/cmd.py:1]: Right now there's a field 
```python
"versions": model._model_config.get("__version__", None)
```

Should we add to that?

[the remark this answers is not in the document]
Comment (isaac-chung) [reply mteb/cmd.py:1]: Let's move this to a separate PR.

File: mteb/evaluation/MTEB.py [hunk 2]
```diff
@@ -285,6 +286,7 @@ def run(
             overwrite_results: Whether to overwrite existing results.
             raise_error: Whether to raise an error if an exception occurs during evaluation.
             co2_tracker: Whether to enable or disable CO2 emissions tracker using codecarbon.
+            model_revision: Revision of the model. If model.model_card_data.base_model_revision is empty, this will be used.
```
[the remark this answers is not in the document]
Comment (isaac-chung) [reply mteb/evaluation/MTEB.py:289]: I was able to get it working at this state, but didn't find success when tickering with the `Encoder` class. See if this aligns with your vision.

File: mteb/evaluation/MTEB.py [hunk 3]
```diff
@@ -297,7 +297,19 @@ def run(
...
+            # check for model revision. For SentenceTransformers, this is available from sentence-transformers==3.0.0.
+            # if not found, check for supplied model revision.
+            if hasattr(model, "revision"):
+                model_revision = model.revision
+            else:
+                try:
+                    model_revision = model.model_card_data.base_model_revision
+                except AttributeError:
+                    logger.warning("Please supply model revision.")
+                    return []
```
Comment (KennethEnevoldsen) [inline mteb/evaluation/MTEB.py:309]: I would just default to "model_name/no_revision_available/results"

**Review by KennethEnevoldsen — APPROVED** [review]

File: mteb/evaluation/MTEB.py (hunk 3 above)
Comment (isaac-chung) [reply mteb/evaluation/MTEB.py]: Added

File: mteb/cmd.py
[the remark this answers is not in the document]
Comment (isaac-chung) [reply mteb/cmd.py:1]: Moved to https://github.com/embeddings-benchmark/mteb/issues/860

**Comment by isaac-chung** [comment]
Added points as well.

**Pull request merged**